NineChronicles.Headless/GraphTypes/ActionMutation.cs

```diff
@@ -19,320 +19,378 @@ public class ActionMutation : ObjectGraphType<NineChroniclesNodeService>
     {
         public ActionMutation()
         {
-            Field<NonNullGraphType<BooleanGraphType>>("createAvatar",
+            Field<NonNullGraphType<TxIdType>>("createAvatar",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<StringGraphType>>
+                    {
+                        Name = "avatarName",
+                        Description = "The character name."
+                    },
+                    new QueryArgument<NonNullGraphType<IntGraphType>>
+                    {
+                        Name = "avatarIndex",
+                        Description = "The index of character slot. 0 ~ 2"
+                    },
+                    new QueryArgument<NonNullGraphType<IntGraphType>>
+                    {
+                        Name = "hairIndex",
+                        Description = "The index of character hair color. 0 ~ 8"
+                    },
+                    new QueryArgument<NonNullGraphType<IntGraphType>>
+                    {
+                        Name = "lensIndex",
+                        Description = "The index of character eye color. 0 ~ 8"
+                    },
+                    new QueryArgument<NonNullGraphType<IntGraphType>>
+                    {
+                        Name = "earIndex",
+                        Description = "The index of character ear color. 0 ~ 8"
+                    },
+                    new QueryArgument<NonNullGraphType<IntGraphType>>
+                    {
+                        Name = "tailIndex",
+                        Description = "The index of character tail color. 0 ~ 8"
+                    }
+                ),
                 resolve: context =>
                 {
-                    try
+                    NineChroniclesNodeService service = context.Source;
+                    PrivateKey privateKey = service.PrivateKey;
+                    BlockChain<NineChroniclesActionType> blockChain = service.Swarm.BlockChain;
+                    var avatarName = context.GetArgument<string>("avatarName");
+                    var avatarIndex = context.GetArgument<int>("avatarIndex");
+                    var hairIndex = context.GetArgument<int>("hairIndex");
```

Comment:
I guess it can be hard for API users because they can't get any other pieces of information for these indices... 🤔